src/libsyntax_ext/deriving/clone.rs

```diff
@@ -80,6 +80,7 @@ pub fn expand_deriving_clone(cx: &mut ExtCtxt,
         additional_bounds: bounds,
         generics: LifetimeBounds::empty(),
         is_unsafe: false,
+        supports_unions: false,
```

Comment:
There are some difficulties with the current implementation of `#[derive(Clone)]`, it generates pattern matching even for the "shallow clone" case to check bounds, but these patterns are not appropriate for unions, this needs to be done in some other way somehow in some other PR. cc @durka 
```
#[derive(Copy, Clone)]
struct S {
	a: u8,
	b: u8,
}
```
=>
```
impl ::std::clone::Clone for S {
    #[inline]
    fn clone(&self) -> S {
        match *self {

            S { a: ref __self_0_0, b: ref __self_0_1 } => { // <== BAD

                ::std::clone::assert_receiver_is_clone(&(*__self_0_0));
                ::std::clone::assert_receiver_is_clone(&(*__self_0_1));
                *self
            }
        }
    }
}
impl ::std::marker::Copy for S { }
struct S {
    a: u8,
    b: u8,
}
```